### DISCUSSION TITLE
CGF File format specification.

### DISCUSSION THREAD

**Discussion by flappybirdace** [Markemp/Cryengine-Converter#133] [Q&A]
I am wondering where the author found specification of the cgf format. Thanks.

**Comment by Markemp** [comment]
Originally there was a file that had some basic Cryengine models that gave some clues about the structure. It's still part of the repo too. But once Cryengine became open source, I was able to get a lot more details about how the files were organized. 

The 010 software package was also invaluable in helping to figure out the structure. It's a pretty neat program and I definitely recommend it if you need to understand the format of binary files.

**Reply by mortany** [reply]
> The 010 software package was also invaluable in helping to figure out the structure. It's a pretty neat program and I definitely recommend it if you need to understand the format of binary files.

Did you try https://kaitai.io/ ?

**Reply by Markemp** [reply]
Just took a look at it, and it's really interesting!  I'll have to dig into the docs to see what their support is for conditional logic.

Doesn't look like they export to C#, unfortunately. :(

**Reply by Kjasi** [reply]
No, but you could write a library in C++ that you use in C#...
